dll/win32/kernel32/client/thread.c

```diff
@@ -327,19 +326,39 @@ CreateRemoteThread(IN HANDLE hProcess,
             {
                 /* Call it instead of going through LPC */
                 Status = CsrCreateRemoteThread(hThread, &ClientId);
-                ASSERT(NT_SUCCESS(Status));
             }
         }
     }
 
-    /* Success */
-    if (lpThreadId) *lpThreadId = HandleToUlong(ClientId.UniqueThread);
+Quit:
+    if (NT_SUCCESS(Status))
```

Comment:
I honestly prefer the old flat code. This if case is only needed, because you removed the ASSERTs above and didn't add checks for failure. Otherwise there is only a single success path and that is good that way. Please fix.

Reply:
Should be better now I think?